pex/pex_builder.py

```diff
@@ -253,7 +254,29 @@ def _add_dist_dir(self, path, dist_name):
         self._copy_or_link(filename, target)
     return CacheHelper.dir_hash(path)
 
+  def _get_installer_paths(self, dist_name):
+    """Set up an overrides dict for WheelFile.install that installs the contents
+    of a wheel into its own base in the pex dependencies cache.
+    """
+    base = os.path.join(self.path(), self._pex_info.internal_cache, dist_name)
+    return {
+      'purelib': os.path.join(base),
+      'headers': os.path.join(base, 'headers'),
+      'scripts': os.path.join(base, 'bin'),
+      'platlib': base,
+      'data': base
+    }
+
   def _add_dist_zip(self, path, dist_name):
+    # We need to distinguish between wheels and other zips. Most of the time,
+    # when we have a zip, it contains its contents in an importable form.
+    # But wheels don't have to be importable, so we need to force them
+    # into an importable shape. We can do that by installing it into its own
+    # wheel dir.
+    if dist_name.endswith("whl"):
+      wf = WheelFile(path)
+      wf.install(overrides=self._get_installer_paths(dist_name), force=True)
```

Comment:
I tested this a bit and noticed that the wheel-installed files will be ignored if they are not *also* written directly to the chroot via `self._chroot.write`. The Chroot class maintains an internal set of written files (by label) in `Chroot.filesets` and files that are included in this data structure get written out to the pex zip in `Chroot.zip` . So any files installed directly by wheel that are not also installed in the normal path below will be ignored.

There doesn't appear to be a clean way to update chroot.filesets without writing directly, so perhaps the best approach here is to install the wheel into a temporary directory, then read the installed RECORD file and write each entry to the chroot?